src/lib.rs

```diff
@@ -178,10 +185,46 @@ impl Client {
         self.conn.flush().await?;
 
         match self.get_read_write_response().await? {
-            Response::Status(s) => Err(s.into()),
-            Response::Data(d) => d.ok_or(Status::NotFound.into()),
-            _ => Err(Status::Error(ErrorKind::Protocol(None)).into()),
+            Response::Status(s) => return Err(s.into()),
+            Response::Data(Some(items)) => {
+                let mut map: HashMap<Vec<u8>, Value> = items
+                    .into_iter()
+                    .map(|item| (item.key.clone(), item))
```

Comment:
This initial `.map()` is required for ownership reasons - we initialize a hashmap that contains a the keys and then `.remove()` them into the results hashmap as we go.

This seemed to be the natural path away form using something like
```
for item in items {
                    if let Some(key) = keys.iter().find(|k| k.as_ref() == item.key) {
                        results.insert(key.clone(), Ok(item));
                    }
                }
```